pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,401 @@
+    def namespaces(self):
+        """
+        Read-only property that returns a list with the names of the
+        namespaces defined for this repository. Note that if there were any
+        leading or trailing slash characters in namespace parameters used
+        to add the namespaces to the repository, they will be removed from
+        the namespaces returned with this property.
+
+        Returns:
+            list of :term:`string` items containing the namespace names
+        """
+        pass
+
+    @abstractmethod
+    def validate_namespace(self, namespace):
+        """
+        Validate if the namespace exists in the repository. If the
+        namespace does not exist a KeyError is raised.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          KeyError: If the namespace is not defined in the repository.
+          ValueError: if the namespace is None.
+        """
+        pass
+
+    @abstractmethod
+    def add_namespace(self, namespace):
+        """
+        Add a CIM namespace to the repository.
+
+        The namespace must not yet exist in the CIM repository.
+
+        The default connection namespace is automatically added to
+        the CIM repository upon creation of this connection.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: If the namespace argument is None or the
+           namespace already exists.
+        """
+        pass
+
+    @abstractmethod
+    def remove_namespace(self, namespace):
+        """
+        Remove a CIM namespace from the repository.
+
+        The namespace must exist in the repository and must be empty.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository. The name is
+            treated case insensitively and it must not be `None`. Any leading
+            and trailing slash characters in the namespace string are ignored
+            when accessing the repository.
+
+        Raises:
+
+          ValueError: Namespace argument is None or the repository namespace
+            is not empty
+          KeyError:  The namespace does not exist in the CIM repository.
+        """
+        pass
+
+    @abstractmethod
+    def get_class_repo(self, namespace):
+        """
+        Get the handle for the data CIM class object store for the namespace
+        provided.
```

Comment:
On "Get the handle": This method returns the store directly, and not a handle to it.
Suggest to change this to "Return the CIM class object store ...".

Reply:
DONE